git_machete/cmd.py

```diff
@@ -2307,59 +2339,60 @@ def print_new_line(new_status: bool) -> None:
                         return
                     flush_caches()
                 elif ans in ('q', 'quit'):
                     return
 
             elif s == UNTRACKED:
-                rems: List[str] = remotes()
-                rmt: Optional[str] = inferred_remote_for_fetching_of_branch(b)
+                rems: List[str] = remotes(cli_ctxt)
+                rmt: Optional[str] = inferred_remote_for_fetching_of_branch(cli_ctxt, b)
                 print_new_line(False)
                 if rmt:
-                    handle_untracked_branch(rmt, b)
+                    handle_untracked_branch(cli_ctxt, rmt, b)
                 elif len(rems) == 1:
-                    handle_untracked_branch(rems[0], b)
+                    handle_untracked_branch(cli_ctxt, rems[0], b)
                 elif "origin" in rems:
-                    handle_untracked_branch("origin", b)
+                    handle_untracked_branch(cli_ctxt, "origin", b)
                 else:
                     # We know that there is at least 1 remote, otherwise 's' would be 'NO_REMOTES'
                     print(fmt(f"Branch `{bold(b)}` is untracked and there's no `{bold('origin')}` repository."))
-                    pick_remote(b)
+                    pick_remote(cli_ctxt, b)
 
-    if opt_return_to == "here":
-        go(initial_branch)
-    elif opt_return_to == "nearest-remaining":
-        go(nearest_remaining_branch)
-    # otherwise opt_return_to == "stay", so no action is needed
+    if cli_ctxt.opt_return_to == "here":
+        go(cli_ctxt, initial_branch)
+    elif cli_ctxt.opt_return_to == "nearest-remaining":
+        go(cli_ctxt, nearest_remaining_branch)
+    # otherwise cli_ctxt.opt_return_to == "stay", so no action is needed
 
     print_new_line(False)
-    status(warn_on_yellow_edges=True)
+    status(cli_ctxt, warn_on_yellow_edges=True)
     print("")
     if cb == managed_branches[-1]:
         msg: str = f"Reached branch {bold(cb)} which has no successor"
     else:
         msg = f"No successor of {bold(cb)} needs to be slid out or synced with upstream branch or remote"
     sys.stdout.write(f"{msg}; nothing left to update\n")
 
-    if opt_return_to == "here" or (opt_return_to == "nearest-remaining" and nearest_remaining_branch == initial_branch):
+    if cli_ctxt.opt_return_to == "here" or (cli_ctxt.opt_return_to == "nearest-remaining" and nearest_remaining_branch == initial_branch):
         print(f"Returned to the initial branch {bold(initial_branch)}")
-    elif opt_return_to == "nearest-remaining" and nearest_remaining_branch != initial_branch:
+    elif cli_ctxt.opt_return_to == "nearest-remaining" and nearest_remaining_branch != initial_branch:
         print(
             f"The initial branch {bold(initial_branch)} has been slid out. Returned to nearest remaining managed branch {bold(nearest_remaining_branch)}")
 
 
-def status(warn_on_yellow_edges: bool) -> None:
+def status(cli_ctxt: CommandLineContext, warn_on_yellow_edges: bool) -> None:
```

Comment:
Probably most of these methods will require all other contexts to be passed as arguments as well...

Reply:
Maybe wrapping then into single unified `Context` with `cli`/`machete`/`git` etc. fields won't be a bad idea

Reply:
sounds good but let's leave the more specific name for now since the other context will be added on the separate PR(s)